x-pack/plugins/task_manager/server/task_store.ts

```diff
@@ -494,6 +499,25 @@ export class TaskStore {
     }
   }
 
+  public async aggregate<TSearchRequest extends AggregationOpts>({
```

Comment:
I think we discussed this earlier, and you had a good reason to define the `WorkloadAggregation` type, but this can be inferred, and you can also remove some of the explicit typing in `workload_statistics.ts`. Here's the diff:

```diff
diff --git a/x-pack/plugins/task_manager/server/monitoring/workload_statistics.ts b/x-pack/plugins/task_manager/server/monitoring/workload_statistics.ts
index 37bf65ba90b..fe70f24684a 100644
--- a/x-pack/plugins/task_manager/server/monitoring/workload_statistics.ts
+++ b/x-pack/plugins/task_manager/server/monitoring/workload_statistics.ts
@@ -9,9 +9,7 @@ import { mergeMap, map, catchError } from 'rxjs/operators';
 import { Logger } from 'src/core/server';
 import { JsonObject } from 'src/plugins/kibana_utils/common';
 import { keyBy, mapValues } from 'lodash';
-import { ESSearchResponse } from '../../../apm/typings/elasticsearch';
 import { AggregatedStatProvider } from './runtime_statistics_aggregator';
-import { ConcreteTaskInstance } from '../task';
 import { parseIntervalAsSecond, asInterval, parseIntervalAsMillisecond } from '../lib/intervals';
 import { AggregationResultOf } from '../../../apm/typings/elasticsearch/aggregations';
 import { HealthStatus } from './monitoring_stats_stream';
@@ -109,7 +107,7 @@ export function createWorkloadAggregator(
 
   return timer(0, refreshInterval).pipe(
     mergeMap(() =>
-      taskStore.aggregate<WorkloadAggregation>({
+      taskStore.aggregate({
         aggs: {
           taskType: {
             terms: { field: 'task.taskType' },
@@ -166,7 +164,7 @@ export function createWorkloadAggregator(
         },
       })
     ),
-    map((result: ESSearchResponse<ConcreteTaskInstance, { body: WorkloadAggregation }>) => {
+    map((result) => {
       const {
         aggregations,
         hits: {
@@ -185,22 +183,13 @@ export function createWorkloadAggregator(
         throw new Error(`Invalid workload: ${JSON.stringify(result)}`);
       }
 
-      const taskTypes = (aggregations.taskType as AggregationResultOf<
-        WorkloadAggregation['aggs']['taskType'],
-        {}
-      >).buckets;
-      const schedules = (aggregations.schedule as AggregationResultOf<
-        WorkloadAggregation['aggs']['schedule'],
-        {}
-      >).buckets;
+      const taskTypes = aggregations.taskType.buckets;
+      const schedules = aggregations.schedule.buckets;
 
       const {
         overdue: { doc_count: overdue },
         scheduleDensity: { buckets: [scheduleDensity] = [] } = {},
-      } = aggregations.idleTasks as AggregationResultOf<
-        WorkloadAggregation['aggs']['idleTasks'],
-        {}
-      >;
+      } = aggregations.idleTasks;
 
       const summary: WorkloadStat = {
         count,
diff --git a/x-pack/plugins/task_manager/server/task_store.ts b/x-pack/plugins/task_manager/server/task_store.ts
index bdeea45ef1d..9941d79b001 100644
--- a/x-pack/plugins/task_manager/server/task_store.ts
+++ b/x-pack/plugins/task_manager/server/task_store.ts
@@ -503,7 +503,7 @@ export class TaskStore {
     aggs,
     query,
     size = 0,
-  }: AggregationOpts) {
+  }: TSearchRequest): Promise<ESSearchResponse<ConcreteTaskInstance, { body: TSearchRequest }>> {
     const { body } = await this.esClient.search<
       ESSearchResponse<ConcreteTaskInstance, { body: TSearchRequest }>
     >({

```

Reply:
Cool! 👍 
I tried and couldn't get it to type check, thanks for the help 😊